tlcc_enclave/enclave/ledger.cpp

```diff
@@ -611,6 +611,13 @@ int parse_endorser_transaction(
                         LOG_ERROR("Ledger: ercc expects only a single write ");
                         valid_tx = false;
                     }
+
+                    // TODO get MRENCLAVE from read/writeset to perform attestation validation once enabled
+                    // this code worked before as we wrote mrenclave during chaincode init; now we write mrenclave
+                    // during setup and thus mrenclave is part of fpc chaincode writeset. This code needs some
+                    // refactoring to resolve this issue; quick and dirty fix to prevent tlcc crashing, this code is
+                    // commented out
+                    /*
```

Comment:
I'm pretty sure the problem that you do not properly identify fpc CC as following code on line 473 in ledger.cpp does not work anymore

            // currently private chaincodes need to have ecc prefix
            if (cc_name.compare(0, 3, "ecc") == 0)
            {
                ecc_namespace = cc_name;
            }
the external-builder version (which you approved ;-) actually does not rename chaincode anymore from foo to ecc_foo! I thought we used that only to remember in the wrapper which CC is fpc CC and forgot that also tlcc relied on this naming convention.
All that said, i think we should leave the commented-out code (although probably with more background as discussed here ...); this code should change sooner than later anyway and re-adding the renaming in the peer wrapper doesn't seem warranted, at least i would hope very much we can do that using this ugly naming convention. That said, in the back of the mind in the past i saw this as a non-problem going away with a proper chaincode type. However, now in v2.0 the chaincode type in the CC-definition does not exist anymore! So it's a good guestion how we are going to handle that. We might at some point revisit the protocol discussion from phoenix (hopefully on updated protocols diagrams reflecting that discussion!) and revisit from whatever we learned in the meantime about fabric v2.0 ...

(Btw: if we really want to keep prefixes, we might use `fpc_` or `fpc-c_` as it would encode the type rather a concept which the user might have no idea about?

Reply:
I defer on this point.

Reply:
added more background as comment in the code.